rules/ast/ast.go

```diff
@@ -29,7 +29,12 @@ import (
 	"github.com/prometheus/prometheus/storage/metric"
 )
 
-var stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+var (
+	stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+	queryTimeout = flag.Duration("query.timeout", 2*time.Minute, "Maximum time a query may take before being aborted.")
+)
+
+var queryTimeoutErr = fmt.Errorf("query timeout after %v", *queryTimeout)
```

Comment:
Should the actual timeout be min(queriedTimeout, defaultTimeout) or would the default be overwritten? This is probably a question of how much we trust the user.

Also, as the timeout logic depends on the stats.TimerGroup anyway would it be reasonable to integrate the timeout into it instead of dragging both values along separately?